app/src/main/java/com/lightricks/feedexercise/network/RetrofitProvider.kt

```diff
@@ -0,0 +1,53 @@
+package com.lightricks.feedexercise.network
+
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import retrofit2.Converter
+import retrofit2.Retrofit
+
+interface RetrofitProvider {
+    val retrofit: Retrofit
+}
+
+/** Entity responsible for providing Retrofit instance
+ *  @param urlProvider provider for base url
+ *  @param loggingInterceptor interceptor for logging requests/responses
+ */
+class RetrofitProviderImpl(
+    private val urlProvider: UrlProvider,
+    private val loggingInterceptor: Interceptor,
+    private val converterFactory: Converter.Factory,
+) : RetrofitProvider {
+
+    private val _retrofit: Retrofit? = null
```

Comment:
Why do we need this explicit backing property?

Reply:
You are right